### DISCUSSION TITLE
Monitor messages on Raspi

### DISCUSSION THREAD

**Discussion by SFSailor** [jvde-github/AIS-catcher#325] [Q&A] [answered]
I have the program running on a Pi Zero W. New at this.  Some questions

- Is there a way to monitor the actual messages coming in? I get counts in the GUI but not the actual messages.
- The Webviewer does not work but gives the message "10.0.0.95 refused to connect."
- I connect to the RTL-SDR v3 on USB when using the GUI, but it says it cannot find the device when using the command line interface...
Any advice?

**Comment by SFSailor** [comment]
Well, I got it working now, and am seeing the values on Putty. But still no webviewer

**Comment by jvde-github** [comment]
Ok, so you are running on a RPI and have the program running (assume starting it via Putty) and you see the NMEA on the screen? 
When you add -N 8100 to the command line (or activate in the GUI) what does the program say when you start it?
You are trying to open the web interface (with the map) from another machine I assume. In the browser, if you go to http://<IP of the Zero:8100 what do you see?

Bit more logs and details would perhaps help to clarify

**Comment by SFSailor** [comment]
When I start via Putty yes, I see NMEA on the screen.

Starting via gui from my windoze laptop, here is the log:
`This is free software; see the source for copying conditions.There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Detached kernel driver
Found Rafael Micro R820T tuner
TCP feed: open socket for host: aiscatcher.org, port: 4242, filter: OFF, PERSIST: ON, KEEP_ALIVE: OFF, JSON: ON, status: pending
Allocating 15 zero-copy buffers
Device    : RTL2838UHIDIR
Settings  : rate 1536K bw 192K format CU8 tuner AUTO rtlagc ON biastee OFF buffer_count 24
Model #0 -> (Src: 1, Grp: 1): [AIS engine v0.60] ps_ema ON afc_wide ON droop ON fp_ds OFF dsk OFF
TCP (aiscatcher.org:4242): connected to server.`

That is at xx.xx.xx.xx:8110 which all works fine except for the webviewer tab.  That SHOULD give me a map or chart and instead gives me a little sad face and eventually the "refused to connect" message.

I am running on a raspi zero W, using the precompiled debian distribution as the program won't compile on my device, stalling at 40%

**Reply by jvde-github** [reply, marked as answer]
Ok, I suspect the webviewer is not switched on. Can you go to Output -> Webserver:
<img src="https://github.com/user-attachments/assets/5fb7e83d-8fc2-4e6b-9ad0-cbf865246d32" alt="Webserver Output" width="10%">

Ensure that the webserver is active at port 8100:
<img src="https://github.com/user-attachments/assets/c7b86efb-6846-435e-836e-19e8d97194bb" alt="Webserver Active" width="30%">

Then scroll to the bottom and press Save to store the new settings. And then go back to Control to Restart the program. You should then see something in the log like:
<img src="https://github.com/user-attachments/assets/ec6f42c1-7cba-4521-a65f-3ac486f7a935" alt="Log Output" width="50%">

If that is the case, you will have the webviewer at the same location as the GUI but then at port 8100.

Is that perhaps the issue? The webserver does not run by default.

**Reply by SFSailor** [reply]
Well, that was precisely it.  I did not realize it needed to be on.  It's working now and I am getting a very nice set of signals from a very slapdash installation!  On to the next issue!
